### DISCUSSION TITLE
Choosing main .tex file in multifile setting

### DISCUSSION THREAD

**Discussion by TryerGit** [James-Yu/LaTeX-Workshop#3614] [Q&A] [answered]
Hello,

I have been using Vim and VimTeX for my LaTeX needs thus far. I would like to give LaTeX-Workshop and VSCode a try.

Given the following setting:

```tex
% main.tex
\begin{document}
Hello World! \person
\end{document}

% professor.tex
\documentclass{minimal}
\newcommand{\person}{from professor}
\input{main.tex}

% student.tex
\documentclass{minimal}
\newcommand{\person}{from student}
\input{main.tex}
```

can LaTeX-Workshop ask the user which of `professor.tex` or `student.tex` to choose as the "main" file?

VimTeX does provide this option. Please see my discussion [here](https://github.com/lervag/vimtex/issues/2588#issuecomment-1330609427)

Thank you.

----

Sorry for the noise: I should have been more diligent. A similar question was asked and answered in the negative here: https://github.com/James-Yu/LaTeX-Workshop/discussions/3493

**Comment by James-Yu** [comment, marked as answer]
The file in active editor will be used as root if possible. Otherwise, an auto root-detection system will kick in. See https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#multi-file-projects

**Reply by TryerGit** [reply]
Yes, I did go through that document and yet when there are two competing root files, it picks one by itself. I was hoping the user could be prompted for which root to pick if there are two or more equally valid ones.

**Comment by Innokentiy-Alaytsev** [comment]
Hello!

Another potentially related issue is that only `*.tex` files are considered when looking for the root files. It might be unorthodox, but I want my `TiKz` figures to have `*.tikz` extension and sometimes I want to build them on their own. I have added a separate recipe for building the file I want but Latex Workshop is unable to use it because it doesn't find the root file, even if I add the magic comment or add `*.tikz` to `latex-workshop.latex.search.rootFiles.include`
